Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -820,97 +798,73 @@ public Money GetBalance(List<StakeTx> stakeTxes)
-        {
-            this.logger.LogTrace("({0}.{1}:{2},{3}:{4},{5}:{6})", nameof(stakeTxes), nameof(stakeTxes.Count), stakeTxes.Count, nameof(nTargetValue), nTargetValue, nameof(nSpendTime), nSpendTime);
-
-            List<StakeOutput> coins = this.AvailableCoinsForStaking(stakeTxes, nSpendTime);
-            setCoinsRet = new List<StakeTx>();
-            nValueRet = 0;
-
-            foreach (StakeOutput output in coins)
-            {
-                this.logger.LogTrace("Checking if UTXO '{0}/{1}' value {2} can be added.", output.StakeTx.OutPoint.Hash, output.StakeTx.OutPoint.N, output.StakeTx.TxOut.Value);
-                StakeTx pcoin = output.StakeTx;
-
-                // Stop if we've chosen enough inputs.
-                if (nValueRet >= nTargetValue)
-                {
-                    this.logger.LogTrace("Target amount reached, exiting loop.");
-                    break;
-                }
-
-                Money n = pcoin.TxOut.Value;
-                if (n >= nTargetValue)
-                {
-                    // If input value is greater or equal to target then simply insert
-                    // it into the current subset and exit.
-                    setCoinsRet.Add(pcoin);
-                    nValueRet += n;
-                    this.logger.LogTrace("UTXO '{0}/{1}' value {2} can be added and is large enough to reach the target.", output.StakeTx.OutPoint.Hash, output.StakeTx.OutPoint.N, output.StakeTx.TxOut.Value);
-                    break;
-                }
-                else if (n < (nTargetValue + BlockValidator.CENT))
-                {
-                    setCoinsRet.Add(pcoin);
-                    nValueRet += n;
-                    this.logger.LogTrace("UTXO '{0}/{1}' value {2} can be added, we now have {3}.", output.StakeTx.OutPoint.Hash, output.StakeTx.OutPoint.N, output.StakeTx.TxOut.Value, nValueRet);
-                }
-            }
-
-            this.logger.LogTrace("(-):true");
-            return true;
-        }
-
-        private List<StakeOutput> AvailableCoinsForStaking(List<StakeTx> stakeTxes, uint nSpendTime)
+        /// <summary>
+        /// Selects coins that are suitable for staking. 
+        /// <para>
+        /// Such a coin has to be confirmed with enough confirmations - i.e. has suitable depth,
+        /// and it also has to be mature and meet requirement for minimal value.
+        /// </para>
+        /// </summary>
+        /// <param name="stakeTxes">List of coins that are candidates for being used for staking.</param>
+        /// <param name="spendTime">Timestamp of the coinstake transaction.</param>
+        /// <param name="maxValue">Maximal amount of coins that can be used for staking.</param>
+        /// <returns>List of coins that meet the requirements.</returns>
+        private List<StakeTx> SelectSuitableCoinsForStaking(List<StakeTx> stakeTxes, uint spendTime, long maxValue)
```

Comment:
I would actually rename this to FindCoinsForStaking

Reply:
OK, will do. For me this is rather selection than finding, but no problemo.